consumer.go

```diff
@@ -551,22 +579,12 @@ func (bc *brokerConsumer) subscriptionConsumer() {
 		}
 
 		for child := range bc.subscriptions {
-			if err := child.handleResponse(response); err != nil {
-				switch err {
-				case ErrOffsetOutOfRange:
-					// there's no point in retrying this it will just fail the same way again
-					// so shut it down and force the user to choose what to do
-					child.AsyncClose()
-					fallthrough
-				default:
-					child.sendError(err)
-					fallthrough
-				case ErrUnknownTopicOrPartition, ErrNotLeaderForPartition, ErrLeaderNotAvailable:
-					// these three are not fatal errors, but do require redispatching
-					child.trigger <- none{}
-					delete(bc.subscriptions, child)
-					Logger.Printf("consumer/broker/%d abandoned subscription to %s/%d because %s\n", bc.broker.ID(), child.topic, child.partition, err)
-				}
+			child.feeder <- response
+		}
+
+		for i := len(bc.subscriptions); i > 0; i-- {
```

Comment:
A waitgroup would solve 95% of the problem very elegantly, but it doesn't let us communicate back "there was an error, abandon this subscription". However maybe that can be a totally different thing, let me play with this...